admin-dev/themes/new-theme/js/components/multistore-form.ts

```diff
@@ -0,0 +1,115 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ComponentsMap from '@components/components-map';
+import {ConfirmModal} from '@components/modal/confirm-modal';
+
+const {$} = window;
+
+const initMultistoreForm = () => {
+  const MultistoreFormMap = ComponentsMap.multistoreForm;
+  const MultistoreHeaderMap = ComponentsMap.multistoreHeader;
+  const multistoreForm = $(MultistoreFormMap.formRow).closest('form');
+  const $modalItem = $(MultistoreHeaderMap.modal);
+
+  const generateFormValuesHash = () => {
+    const formValues = multistoreForm.serializeArray().reduce((obj: any, item: any) => {
+      const form = multistoreForm[0] as HTMLFormElement;
+      const formElement = form.elements[item.name] as HTMLFormElement;
+
+      if ($(formElement).prop('disabled')) { // Ignore if element is disabled
+        return obj;
+      }
+      const fieldValues = obj;
+
+      if (formElement instanceof HTMLTextAreaElement) {
+        fieldValues[item.name] = item.value
+          .replace(/<[^>]*>?/gm, '') // remove html tags
+          .replace(/\r?\n|\r/g, '') //remove line breaks
+          .replace(/(\.\s)/g, '.'); //remove space after dots added by TinyMCE
+      } else {
+        fieldValues[item.name] = item.value;
+      }
+
+      return fieldValues;
+    });
+
+    return JSON.stringify(formValues);
+  };
+
+  /**
+   * @param {string} path
+   */
+  const showConfirmModal = (
+    path: string,
+  ) => {
+    const confirmTitle = window.translate_javascripts['multistore.modal.confirm_leave.title'];
+    const confirmMessage = window.translate_javascripts['multistore.modal.confirm_leave.body'];
+    const confirmButtonLabel = window.translate_javascripts['multistore.modal.confirm_leave.confirm'];
+    const closeButtonLabel = window.translate_javascripts['multistore.modal.confirm_leave.cancel'];
+    const confirmButtonClass = 'btn-primary';
+
+    const modal = new ConfirmModal(
+      {
+        confirmTitle,
+        confirmMessage,
+        confirmButtonLabel,
+        closeButtonLabel,
+        confirmButtonClass,
+      },
+      () => {
+        window.location.href = path;
+      },
+    );
+
+    modal.show();
+  };
+
+  if (multistoreForm) {
+    const originalFormValuesHash = generateFormValuesHash();
+
+    // Bind click on header's links
+    $modalItem.find('a').each((index, itemLink) => {
+      $(itemLink).on('click', () => {
+        const formValuesHash = generateFormValuesHash();
+
+        if (originalFormValuesHash !== formValuesHash) {
+          const targetUrl = $(itemLink).attr('href');
+          showConfirmModal(`${targetUrl}`);
```

Comment:
What if the `<a>` comes from a module, doesn't have an href because it's wrong and should use a button?

What about buttons inside the header? In case a module adds a button + some JS

```suggestion
          if (targetUrl && typeof targetUrl === 'string') {
            showConfirmModal(targetUrl);
          }
```

So you use typescript types